src/cross_section.F90

```diff
@@ -526,6 +577,192 @@ pure function find_energy_index(mat, E) result(i)
+    complex(8) :: c_temp   ! complex temporary variable
+    complex(8) :: w_val    ! The faddeeva function evaluated at Z
+    complex(8) :: Z        ! sqrt(atomic weight ratio / kT) * (sqrt(E) - pole)
+    real(8) :: sqrtE       ! sqrt(E), eV
+    real(8) :: invE        ! 1/E, eV
+    real(8) :: dopp        ! sqrt(atomic weight ratio / kT)
+    real(8) :: dopp_ecoef  ! sqrt(atomic weight ratio * pi / kT) / E
+    real(8) :: temp        ! real temporary value
+    real(8) :: E           ! energy, eV
+    real(8) :: sqrtkT      ! sqrt(kT (in eV))
+    integer :: iP          ! index of pole
+    integer :: iC          ! index of curvefit
+    integer :: iW          ! index of window
+    integer :: startw      ! window start pointer (for poles)
+    integer :: startw_1    ! window start pointer - 1
+    integer :: startw_endw ! window start pointer - window end pointer
+    integer :: endw        ! window end pointer
+
+    ! Convert to eV
+    E = Emev * 1.0e6_8
+    sqrtkT = sqrtkT_ * 1.0e3_8
+
+    sqrtE = sqrt(E)
+    invE = ONE/E
+
+    if(.not. mp_already_alloc) then
+      call multipole_eval_allocate()
+    end if
+
+    ! Locate us
+    iW = floor((sqrtE - sqrt(multipole % start_E))/multipole % spacing + ONE)
+
+    startw = multipole % w_start(iW)
+    startw_1 = startw - 1 ! This is an index shift parameter.
+    endw = multipole % w_end(iW)
+    startw_endw = endw - startw + 1
+
+    ! Fill in factors
+    if (startw <= endw) then
+      call fill_factors(multipole, sqrtE, sigT_factor, twophi, multipole % num_l)
+    end if
+
+    ! Generate some doppler broadening parameters
+
+    ! dopp_ecoef is inverse of dopp, divided by E, multiplied by sqrt(pi).
+    dopp = multipole % sqrtAWR / sqrtKT
+    dopp_ecoef = dopp * invE * SQRT_PI
+
+    sigT = ZERO
+    sigA = ZERO
+    sigF = ZERO
+    ! Evaluate linefit first
+
+    if(sqrtkT /= 0 .and. multipole % broaden_poly(iW) == 1) then ! Broaden the curvefit.
+      call broaden_n_polynomials(E, dopp, multipole % fit_order + 1, broadened_polynomials)
+
+      do iC = 1, multipole % fit_order+1
+        sigT = sigT + multipole % curvefit(FIT_T, iC, iW)*broadened_polynomials(iC)
+        sigA = sigA + multipole % curvefit(FIT_A, iC, iW)*broadened_polynomials(iC)
+        if (multipole % fissionable) then
+          sigF = sigF + multipole % curvefit(FIT_F, iC, iW)*broadened_polynomials(iC)
+        end if
+      end do
+    else ! Evaluate as if it were a polynomial
+      temp = invE
+      do iC = 1, multipole % fit_order+1
+
+        sigT = sigT + multipole % curvefit(FIT_T, iC, iW)*temp
+        sigA = sigA + multipole % curvefit(FIT_A, iC, iW)*temp
+        if (multipole % fissionable) then
+          sigF = sigF + multipole % curvefit(FIT_F, iC, iW)*temp
+        end if
+
+        temp = temp * sqrtE
+      end do
+    end if
+
+    ! Then get the poles we want and broaden them.
+
+    if (sqrtkT == ZERO) then
+      ! If at 0K, use asymptotic form.
+      do iP = startw, endw
```

Comment:
We usually use all lower case for variables.  Replace `iP` with `i_p`, or even better, `i_pole`